File: 2022/Day01/Main.hs
import System.IO
import Data.List

sumGroups :: [String] -> [Int] -> [Int]
sumGroups [] ns = ns
sumGroups ("":xs) ns = sumGroups xs (0:ns)
sumGroups (x:xs) (n:ns) = sumGroups xs ((n + read x):ns)

main = do
  handle <- openFile "input.txt" ReadMode
  contents <- hGetContents handle
  let outputtop = sort (sumGroups (lines contents) [0])
  print (sum (take 1 (reverse (outputtop))))
  print (sum (take 3 (reverse (outputtop))))
  hClose handle

File: 2022/Day01/main.py
# Advent of Code 2022 - Day 1
# Finding the elf with the most calories

# Import the File
with open('input.txt') as f:
  lines = [line.rstrip() for line in f]

totals = [0]

i = 0
for line in lines:
  if line != "":
    totals[i] += int(line)
  else:
    totals.append(0)
    i += 1

totals.sort(reverse=True)

print(totals[0])
print(totals[0]+totals[1]+totals[2])
